File: desktop/src/com/inkostilation/pong/desktop/network/ServerThread.java
package com.inkostilation.pong.desktop.network;

import java.io.IOException;

public class ServerThread extends Thread {

    private boolean active = false;
    private boolean finished = false;

    private static ServerThread instance;

    public static ServerThread getInstance() {
        if (instance == null) {
            instance = new ServerThread();
        }
        return instance;
    }

    @Override
    public synchronized void start() {
        active = true;
        super.start();
    }

    @Override
    public void run() {
        while (active) {
            Network.getEngine().act(0);
        }

        // last messages to disconnect and etc.
        try {
            Network.getEngine().quit(null);
        } catch (IOException e) {
            e.printStackTrace();
        }

        finished = true;
    }

    public void stopServer() {
        this.active = false;
    }

    public boolean isFinished() {
        return finished;
    }
}

File: desktop/src/com/inkostilation/pong/desktop/network/ServerThread.kt
package com.inkostilation.pong.desktop.network

import java.io.IOException

class ServerThread : Thread() {
    private var active = false
    var isFinished = false
        private set

    @Synchronized
    override fun start() {
        active = true
        super.start()
    }

    override fun run() {
        while (active) {
            Network.getEngine().act(0f)
        }
        // last messages to disconnect and etc.
        try {
            Network.getEngine().quit(null)
        } catch (e: IOException) {
            e.printStackTrace()
        }
        isFinished = true
    }

    fun stopServer() {
        active = false
    }

    companion object {
        var instance: ServerThread? = null
            get() {
                if (field == null) {
                    field = ServerThread()
                }
                return field
            }
            private set
    }
}